wagtail/admin/viewsets/model.py

```diff
@@ -35,15 +35,28 @@ class ModelViewSet(ViewSet):
     def permission_policy(self):
         return ModelPermissionPolicy(self.model)
 
+    def get_index_view_kwargs(self):
+        return {
+            "model": self.model,
+            "permission_policy": self.permission_policy,
+            "index_url_name": self.get_url_name("index"),
+            "index_results_url_name": self.get_url_name("index_results"),
+            "add_url_name": self.get_url_name("add"),
+            "edit_url_name": self.get_url_name("edit"),
+            "header_icon": self.icon,
+        }
+
```

Comment:
This provides an easier way to override a few kwargs for the view, I wonder if we should do the same for other views in the viewset (and refactor `SnippetViewSet` to make use of it). In a separate PR, of course

Reply:
I think there's a lot we could do to improve the maintainability and reduce the verbosity of viewsets as a pattern. My biggest bugbear with it is that the knowledge of "which parameters are accepted by the view, and thus _must_ be passed from the viewset" is duplicated in an error-prone way. We can't necessarily refer to the view to get a neat list of attributes, because they might be coming in through mixins.

Maybe this calls for some sort of `Meta` class on the view to declare a set of attributes that we allow / expect to be supplied by the viewset, with the appropriate metaclass magic to merge those definitions if they're defined on mixins. The viewset would then consult that definition and pass on the appropriate subset of its own attributes (with some provision for mapping non-matching names, like `icon` vs `header_icon` above). And while I'm putting together my wishlist, it would be really neat if it worked for overriding methods too - so, for example, a ChooserViewSet subclass could define a `get_display_title` method once, and the viewset mechanism can then discover which of its views implement that method and allow it to be overridden, and dynamically create subclasses of those views with the relevant overrides.

I could easily see this becoming a separate utility package independent of Wagtail.